### PR TITLE
#645 Add HDEL command

### PR DESCRIPTION
**Pull request by manishrw**
HDEL command implementation:
- Added input validation for minimum number of arguments
- Implemented deletion of multiple fields from a hash
- Added type checking to ensure the key is a hash
- Returns the count of successfully deleted fields
- Updates the store only if fields were actually deleted

### CODE REVIEW

**Comment by shardul08** [comment]
@manishrw please resolve the conflicts

**Comment by manishrw** [comment]
@shardul08 conflicts have been resolved. Let's review it before any new conflicts arise.

**Comment by manishrw** [comment]
@arpitbbhayani  Can you approve if it looks good?
Let's merge it before there are any further conflicts.

**Comment by manishrw** [comment]
@lucifercr07 @JyotinderSingh Can you review and approve if it looks good?

File: internal/eval/commands.go
Comment (lucifercr07) [inline internal/eval/commands.go:1]: @manishrw thanks for contributing, you please add integration tests also.

Comment (manishrw) [reply internal/eval/commands.go:1]: have added integration test

**Review by lucifercr07 — APPROVED** [review]
LGTM, thanks for contributing.

**Pull request merged**